cordova-lib/src/cordova/plugin.js

```diff
@@ -157,6 +157,7 @@ module.exports = function plugin(command, targets, opts) {
 
                             if (missingVariables.length) {
                                 shell.rm('-rf', pluginInfo.dir);
+                                events.emit('verbose', 'Removing ' + pluginInfo.dir + ' due to installation failure');
                                 var msg = 'Variable(s) missing (use: --variable ' + missingVariables.join('=value --variable ') + '=value).';
```

Comment:
I now see what you mean. Could you please move the ```events.emit('Removing ...')``` to L159 right before the removal actually occurs ? I hadn't notice that and got confused.
I think we should also emit a message about the variables missing.(```events.emit(msg)```)